src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -612,13 +612,34 @@ public void AppendConflictingField(EnumFieldInfo other)
                 // Walk the existing list to ensure we do not add duplicates.
                 foreach (EnumFieldInfo conflictingField in conflictingFields)
                 {
-                    if (conflictingField.Kind is EnumFieldNameKind.Default || conflictingField.JsonName.Equals(other.JsonName, StringComparison.Ordinal))
+                    if (ConflictsWith(conflictingField, other))
                     {
                         return;
                     }
                 }
 
                 conflictingFields.Add(other);
+
+                // Determines whether the first field info matches everything that the second field info matches,
+                // in which case the second field info is redundant and doesn't need to be added to the list.
+                static bool ConflictsWith(EnumFieldInfo current, EnumFieldInfo other)
```

Comment:
Nit: XML doc comments could be used here

Reply:
Doc comments don't seem to be supported for local functions - they don't show up in quickinfo/completion for local functions and VS doesn't auto-complete them.